### DISCUSSION TITLE
Planning engine terminating due to segmentation violation

### DISCUSSION THREAD

**Discussion by sgfisme** [frePPLe/frepple#528] [Q&A]
I got the error as 'Planning engine terminating due to segmentation violation' after I changed the source item's type from null to 'make to order'. I rolled back my data injection that the source item's type as null and planned successfully.

My data statistics as below:
Demand: 1,185
Item: 110,431
Operation: 472,309
Operation Material: 89,614
Manufacturing Order: 21,746
Planning Clusters: 14792

Those statistics are not different whether source item types changed or not. But the engine execution time spent over 6 hrs and failed after source item's type changed as null. What's different the item's type as 'make to order' and null?

**Comment by jdetaeye** [comment]
See this page with examples on when & how to use the item type field https://frepple.org/docs/current/examples/buffer/make-to-order.php
The type has a very important impact on the model structure, size and the behavior when used.

The crash is not expected when you change the item type. If it's reproducible in a small dataset and with a dataset that correctly uses the batch field, this could be a bug.

**Reply by sgfisme** [reply]
@jdetaeye I tried to reproduce in 2 small data sets correctly using the batch field:
1. Keep priority 1 orders to plan. They were 254 orders and the engine ran successfully.
2. Keep 16 orders on 2 clusters and ran successfully.

For scenario 2, the engine crashed over 16 orders on 2 clusters. Any possible for the engine crashed because plan orders at the same time on the cluster had a complex route that had more alternative routes?

**Comment by jdetaeye** [comment]
@sgfisme   With the provided information it's absolutely impossible to understand what's the root of your crash.

**Reply by sgfisme** [reply]
@jdetaeye Sorry. I will try to describe my question more specific.

I read your reference as https://frepple.org/docs/current/examples/buffer/make-to-order.php and our supply path seems like the config-to-order example. If we set item type as 'make to stack' or null on subassembly, could we set item type as 'make to order' on the source item as Component-3 for subassembly?

![make-to-order-1](https://user-images.githubusercontent.com/44051141/171302352-e9216a5e-c263-436d-86c8-faf324cdc966.png)

**Comment by jdetaeye** [comment]
>> If we set item type as 'make to stack' or null on subassembly, could we set item type as 'make to order' on the source item as Component-3 for subassembly?

Yes, that's possible.   Such a configuration is called "serialized production" on the same page  https://frepple.org/docs/current/examples/buffer/make-to-order.php.   
The batches for the component will be automatically generated by frepple. It's not propagated backward from the sales order any longer, as the make-to-stock item in the chain interrupts that propagation.
